servers/rendering/renderer_rd/storage_rd/material_storage.cpp

```diff
@@ -987,25 +987,16 @@ void MaterialStorage::MaterialData::free_parameters_uniform_set(RID p_uniform_se
 }
 
 bool MaterialStorage::MaterialData::update_parameters_uniform_set(const HashMap<StringName, Variant> &p_parameters, bool p_uniform_dirty, bool p_textures_dirty, const HashMap<StringName, ShaderLanguage::ShaderNode::Uniform> &p_uniforms, const uint32_t *p_uniform_offsets, const Vector<ShaderCompiler::GeneratedCode::Texture> &p_texture_uniforms, const HashMap<StringName, HashMap<int, RID>> &p_default_texture_params, uint32_t p_ubo_size, RID &uniform_set, RID p_shader, uint32_t p_shader_uniform_set, bool p_use_linear_color, bool p_3d_material) {
-	if ((uint32_t)ubo_data.size() != p_ubo_size) {
-		p_uniform_dirty = true;
-		if (uniform_buffer.is_valid()) {
-			RD::get_singleton()->free(uniform_buffer);
-			uniform_buffer = RID();
-		}
-
-		ubo_data.resize(p_ubo_size);
-		if (ubo_data.size()) {
-			uniform_buffer = RD::get_singleton()->uniform_buffer_create(ubo_data.size());
-			memset(ubo_data.ptrw(), 0, ubo_data.size()); //clear
-		}
+	p_uniform_dirty = true;
```

Comment:
Always dirty uniform?

Then:

https://github.com/godotengine/godot/pull/90284/files#diff-89b51fc83d56c557d031b1750f010a1846b685ceb9f13ceb71d431e349aace2cR1003-R1006

Reply:
Well spotted! Thanks!

This appears to be a simple merge error. However I am a bit hesitant because suspiciously this bug would have hided another bug that was fixed by TheForge. I'll have to look into this more deeply.